File: 3152-MaximumValueOfAnOrderedTripletIi/3152-MaximumValueOfAnOrderedTripletIi.py
# Last updated: 6/11/2026, 11:19:55 AM
class Solution:
    def maximumTripletValue(self, nums: List[int]) -> int:
        store = []
        maxright = [0]*(len(nums)+1)
        for i in range(len(nums)-2,-1,-1):
            maxright[i] = max(maxright[i+1],nums[i+1])
        # print(maxright)
        maxv = 0
        # nums2 = [i]*len(nums)
        for i in range(len(nums)-2):
            for j in range(i+1,len(nums)-1):
                if(nums[j]>=nums[i]):
                    break
                maxv = max((nums[i]-nums[j])*maxright[j],maxv)
             
                
        return maxv

File: 3152-MaximumValueOfAnOrderedTripletIi/3152-MaximumValueOfAnOrderedTripletIi.cpp
// Last updated: 6/11/2026, 11:19:47 AM

using ll = long long;
class Solution {
public:
    long long maximumTripletValue(vector<int>& num) {
        ll n = num.size();
        vector<ll> nums;
        for(int el: num){
            nums.push_back(el);
        }
        
        vector<ll> maxvl(n, -1e18);
        vector<ll> maxvr(n, -1e18);
        ll ans = 0;
        // for(ll i=1; i<n; i++){
        //     maxvl[i] = max(maxvl[i-1], nums[i-1]);
        // }
        for(ll i=n-2; i>=0; i--){
            maxvr[i] = max(maxvr[i+1], nums[i+1]);
        }
        for(ll i=1; i<n-1; i++){
            maxvl[i] = max(maxvl[i-1], nums[i-1]);
            ans = max(ans, (maxvl[i]-nums[i])*(maxvr[i]));
        }
        return ans;
    }
};